src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final int OPERATION_INDEX = 0;
+    private static final int AMOUNT_INDEX = 1;
+    private static final int REQUIRED_PARTS_LENGTH = 2;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            int totalSupply = 0;
+            int totalBuy = 0;
+            String line;
+
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.trim().split(",");
+                if (parts.length == REQUIRED_PARTS_LENGTH) {
+                    String operation = parts[OPERATION_INDEX];
+                    int amount = Integer.parseInt(parts[AMOUNT_INDEX]);
+
+                    if (SUPPLY.equals(operation)) {
+                        totalSupply += amount;
+                    } else if (BUY.equals(operation)) {
+                        totalBuy += amount;
+                    }
+                }
+            }
```

Comment:
Performing business logic while keeping open connection to file is generally not a best practice.
Please read file then perform calculations and then write result to file.
After logic is separated please create private method for each part and in public method just call these 3 methods to make code more readable